### ISSUE TITLE
CKV_AZURE_43 false-positive alert

### ISSUE THREAD

**Issue by iddqdiddqd** [bridgecrewio/checkov#4467]
**Describe the issue**
Existing check `CKV_AZURE_43` is being triggered with false-positive alert in my case.

**Examples**
Current `azurerm_storage_account` name I have in the code:
``` terraform
name = substr("${var.environment}${var.name}mlstorage${random_string.postfix.result}", 0, 24)
```
random string settings:
``` terraform
resource "random_string" "postfix" {
  length  = 6
  special = false
  upper   = false
}
```

The `environment` and `name` variables consist of lowercase letters only. Thus, the `CKV_AZURE_43` check failure should not be occurring.

**Version (please complete the following information):**
``` bash
checkov --version
2.2.344
```

**Comment by askpatrickw** [comment]
I also see this on v2.3.247 (latest available via Brew)

**Comment by danpetitt** [comment]
I dont understand what this is supposed to warn for; storage accounts can only use lowercase letters and digits; so what rules are we supposed to adhere to?
